admin/class-papercups-admin.php

```diff
@@ -83,51 +70,175 @@ public function enqueue_styles()
-		 *
-		 * An instance of this class should be passed to the run() function
-		 * defined in Papercups_Loader as all of the hooks are defined
-		 * in that particular class.
-		 *
-		 * The Papercups_Loader will then create the relationship
-		 * between the defined hooks and the functions defined in this
-		 * class.
-		 */
-
-		wp_enqueue_script($this->plugin_name, plugin_dir_url(__FILE__) . 'js/papercups-admin.js', array('jquery'), $this->version, false);
+	public function enqueue_scripts(){
+		wp_enqueue_script($this->plugin_name, plugin_dir_url(__FILE__) . 'js/papercups-admin.js', array('jquery', 'wp-color-picker'), $this->version, false);
 	}
 
-	/**
-	 * Register the administration menu for the plugin into the WordPress Dashboard menu.
-	 *
-	 * @since    1.0.0
-	 */
-	public function add_plugin_admin_menu()
-	{
-		/*
-		 * Add a settings page for this plugin to the Settings menu.
-		 */
-		add_menu_page(
-			__('Papercups', $this->plugin_name),
-			__('Papercups', $this->plugin_name),
-			'manage_options',
-			$this->plugin_name,
-			array($this, 'display_plugin_admin_page'),
-			'dashicons-admin-comments'
-		);
-	}
+  function init_settings() {
+    $this->register_settings();
 
-	/**
-	 * Render the settings page for this plugin.
-	 *
-	 * @since    1.0.0
-	 */
-	public function display_plugin_admin_page()
-	{
-		require plugin_dir_path(dirname(__FILE__)) . 'admin/partials/papercups-admin-display.php';
-	}
+    settings_fields('papercups_widget_setting');
+    do_settings_fields('papercups_account_id', 'papercups_widget_setting');
+    add_settings_section(
+        'papercups_settings_section',
+        'Papercups Settings Section',
+        array($this, 'settings_section_cb'),
+        'general'
+    );
+
+    $this->add_settings_fields();
+  }
+
+  private function register_settings() {
+    register_setting('general', 'papercups_account_id');
+    register_setting('general', 'papercups_widget_title');
+    register_setting('general', 'papercups_widget_subtitle');
+    register_setting('general', 'papercups_new_message_placeholder');
+    register_setting('general', 'papercups_greeting');
+    register_setting('general', 'papercups_primary_color');
+    register_setting('general', 'papercups_require_email_upfront');
+    register_setting('general', 'papercups_base_url', array('default' => 'https://app.papercups.io'));
+  }
+
+  private function add_settings_fields() {
+    add_settings_field(
+      'papercups_account_id',
+      __('Papercups accound ID', $this->plugin_name),
+      array($this, 'text_input_callback'),
+      'general',
+      'papercups_settings_section',
+      array(
+        'type'      => 'text',
+        'value'     => get_option('papercups_account_id'),
+        'class'     => 'regular-text ltr',
+        'label_for' => 'papercups_account_id',
+        'tip'       => __('You can your account ID on papercups app dashboard', $this->plugin_name)
+      ),
+    );
+
+    add_settings_field(
```

Comment:
is there a way to mark certain fields optional? mosts of these fields will likely not need to be set, or we can just prefill them with default values

Reply:
yes sure we can prefill them by adding the default `defalut` value like here
`register_setting('general', 'papercups_base_url', array('default' => 'https://app.papercups.io'));`
if you have any suggestions we can add other defaults